test/openssl/test_pkey.rb

```diff
@@ -244,4 +244,210 @@ def test_to_text
     rsa = Fixtures.pkey("rsa1024")
     assert_include rsa.to_text, "publicExponent"
   end
+
+  if openssl?(3, 0, 0)
+    def test_from_parameters_with_invalid_alg
+      e = assert_raise(OpenSSL::PKey::PKeyError) {
+        OpenSSL::PKey.from_parameters("ASR", {})
+      }
+      assert_match(/^EVP_PKEY_CTX_new_from_name: unsupported/, e.message)
+    end
+
+    def test_s_from_parameters_rsa_with_n_e_and_d_given_as_integers
+      new_key = OpenSSL::PKey.from_parameters("RSA", n: 3161751493,
+                                                     e: 65537,
+                                                     d: 2064855961)
```

Comment:
Is it intentional that you are using mixture of the symbol key `n:` in this test and String key `"n" =>` in another test?

Reply:
The intention was to test that keys could also be passed as strings. Maybe it would be clearer to have a dedicated test for that instead of testing it "on the side"?

Reply:
Yes, I think the dedicated test for testing the symbol keys is a good idea to show the intention clearly. Checking the existing `test/openssl/test_pkey.rb`, there are the cases using string keys for a hash in the file. So, in my opinion, maybe it's better to use the String keys as a default, and a dedicated test for testing the symbol keys.

Reply:
Now tests are using string keys, except one dedicated test with symbol keys